Feature reddit

tetris-11 opened the pull request:
Hey, really love your app - thought I would have a go at extending functionationality to reddit subs ( e.g. https://imgur.com/r/mountains )

I've got it to work:
   * Added extra reddit 'mode'
   * Added a default subreddit TextField in Settings (saved)
   * Added another subreddit TextField in the Main page (shown only in reddit mode)(not saved)
   * Added a Day/Week/Month/Year/All sorting box (shown in reddit mode).
   
Bugs:
     - I can't seem to anchor the subreddit TextField on the Main page properly.
     - The 'reddit' mode needs to be reselected on startup for it to actually work

I tried to keep it as true to your existing model as possible, but please take what you like and leave what you don't.

If you're not planning to do anything soon, would it be okay if I share my modified RPM on Maemo?

walokra commented:
Thanks for the PR, it looks mostly good. I'll see to the issues you mentioned and publish new version probably later this week at least to OpenRepos.

Does the RPM for Sailfish OS work or does it need something more? If it does then you can share your modified RPM.

tetris-11 commented:
Oh awesome, thanks!

The RPM works, but it's still a little fiddly and the subreddit search field makes the UI look a bit unsightly.

Also I'm not happy about some of the changes I added in *imgur.js*:
      
- Essentially it tries to fetch gallery images the regular way, but I added a hook so that upon a '404' response it then tries the Reddit method of fetching gallery images.
- This is needless overhead. I could have simply tested that `settings.mode_reddit` was active and not needed to add the hook at all.
- Also no album functionality under Reddit mode, as I've just discovered.

I think I'll wait for your more comprehensive update so that I don't push out an inferior package
